injectproxy/silences.go

```diff
@@ -35,21 +36,76 @@ import (
 func (r *routes) silences(w http.ResponseWriter, req *http.Request) {
 	switch req.Method {
 	case "GET":
-		r.enforceFilterParameter(w, req)
+		r.passthrough(w, req)
 	case "POST":
 		r.postSilence(w, req)
 	default:
 		http.NotFound(w, req)
 	}
 }
 
+func (r *routes) filterSilences() func(*http.Response) error {
+	return func(resp *http.Response) error {
+		if resp.Request.Method == http.MethodPost {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			// Pass non-200 responses as-is.
+			return nil
+		}
+
+		apir, err := getAlertmanagerAPIResponse(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, "can't decode API response")
```

Comment:
Would you mind rebasing this. The `github.com/pkg/errors` errors package is deprecated and archived. You can now do this:

```suggestion
			return fmt.Errorf("can't decode API response: %w", err)
```